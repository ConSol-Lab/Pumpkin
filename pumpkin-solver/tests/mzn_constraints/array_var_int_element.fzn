var 1..3: index :: output_var;
var 1..6: rhs :: output_var;

var 1..2: v1 :: output_var;
var 2..3: v2 :: output_var;
var 5..6: v3 :: output_var;

constraint array_var_int_element(index, [v1, v2, v3], rhs);

solve satisfy;
