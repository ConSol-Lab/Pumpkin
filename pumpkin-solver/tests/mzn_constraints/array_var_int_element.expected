index = 1;
rhs = 1;
v1 = 1;
v2 = 2;
v3 = 5;
----------
index = 2;
rhs = 2;
v1 = 1;
v2 = 2;
v3 = 5;
----------
index = 3;
rhs = 5;
v1 = 1;
v2 = 2;
v3 = 5;
----------
index = 1;
rhs = 2;
v1 = 2;
v2 = 2;
v3 = 5;
----------
index = 2;
rhs = 2;
v1 = 2;
v2 = 2;
v3 = 5;
----------
index = 3;
rhs = 5;
v1 = 2;
v2 = 2;
v3 = 5;
----------
index = 1;
rhs = 1;
v1 = 1;
v2 = 3;
v3 = 5;
----------
index = 2;
rhs = 3;
v1 = 1;
v2 = 3;
v3 = 5;
----------
index = 3;
rhs = 5;
v1 = 1;
v2 = 3;
v3 = 5;
----------
index = 1;
rhs = 2;
v1 = 2;
v2 = 3;
v3 = 5;
----------
index = 2;
rhs = 3;
v1 = 2;
v2 = 3;
v3 = 5;
----------
index = 3;
rhs = 5;
v1 = 2;
v2 = 3;
v3 = 5;
----------
index = 1;
rhs = 1;
v1 = 1;
v2 = 2;
v3 = 6;
----------
index = 2;
rhs = 2;
v1 = 1;
v2 = 2;
v3 = 6;
----------
index = 3;
rhs = 6;
v1 = 1;
v2 = 2;
v3 = 6;
----------
index = 1;
rhs = 2;
v1 = 2;
v2 = 2;
v3 = 6;
----------
index = 2;
rhs = 2;
v1 = 2;
v2 = 2;
v3 = 6;
----------
index = 3;
rhs = 6;
v1 = 2;
v2 = 2;
v3 = 6;
----------
index = 1;
rhs = 1;
v1 = 1;
v2 = 3;
v3 = 6;
----------
index = 2;
rhs = 3;
v1 = 1;
v2 = 3;
v3 = 6;
----------
index = 3;
rhs = 6;
v1 = 1;
v2 = 3;
v3 = 6;
----------
index = 1;
rhs = 2;
v1 = 2;
v2 = 3;
v3 = 6;
----------
index = 2;
rhs = 3;
v1 = 2;
v2 = 3;
v3 = 6;
----------
index = 3;
rhs = 6;
v1 = 2;
v2 = 3;
v3 = 6;
----------
==========
